controller/api/destination/watcher/endpoints_watcher_test.go

```diff
@@ -851,6 +851,148 @@ status:
 			expectedError:                    false,
 			expectedLocalTrafficPolicy:       true,
 		},
+		{
```

Comment:
can we also add tests that ensure that we pick only the first address. And maybe if we are so intentional, we should also warn that we pick the first update only (in case there is more than one...).